lib/types/src/memory_view.rs

```diff
@@ -64,6 +64,33 @@ where
             _phantom: PhantomData,
         }
     }
+
+    /// Creates a subarray view from this MemoryView.
+    pub fn subarray(&self, start: u32, end: u32) -> Self {
```

Comment:
We can probably use a https://doc.rust-lang.org/std/ops/struct.Range.html here and be able to do it  as `x.subarray(start..end)`; or some generic type using https://doc.rust-lang.org/std/ops/trait.RangeBounds.html for full flexibility

Reply:
Done!